### ISSUE TITLE
PageableDefault is not reported correctly

### ISSUE THREAD

**Issue by eiswind** [springdoc/springdoc-openapi#1415]
**Describe the bug**
Pageable is documented to be used with Parameter(hidden) and PageableAsQueryParam.

When we add PageableDefault, the default page size should be 10 but is reported as 20.

**To Reproduce**
- What version of spring-boot you are using?
- 2.6.2
- What modules and versions of springdoc-openapi are you using?
- 1.6.3
- What is the actual and the expected result using OpenAPI Description (yml or json)?
actual
paths:
  /api:
    get:
      tags:
      - test-controller
      operationId: findAll
      parameters:
      - name: page
        in: query
        description: Zero-based page index (0..N)
        schema:
          type: integer
          default: 0
      - name: size
        in: query
        description: The size of the page to be returned
        schema:
          type: integer
          default: 20
     

expected
paths:
  /api:
    get:
      tags:
      - test-controller
      operationId: findAll
      parameters:
      - name: page
        in: query
        description: Zero-based page index (0..N)
        schema:
          type: integer
          default: 0
      - name: size
        in: query
        description: The size of the page to be returned
        schema:
          type: integer
          default: 10
     

- Provide with a sample code (HelloController) or Test that reproduces the problem
https://github.com/eiswind/openapi-pageable

**Expected behavior**
I expect d a default of 10. I guess we could have the same issue as well with the sort parameter.

**Comment by bnasslahsen** [comment]
@eiswind, 

You are having wrong settings.  All is explained in the [documentation](https://springdoc.org/#how-can-i-map-pageable-spring-date-commons-object-to-correct-url-parameter-in-swagger-ui).

To enable `PageableDefault` support:
- Add:

```pom.xml
<dependency>
	<groupId>org.springdoc</groupId>
	<artifactId>springdoc-openapi-data-rest</artifactId>
	<version>1.6.3</version>
</dependency>
```

And in your controller:

```java
  @GetMapping(params = "page")
  List<TestEntity> findAll(@PageableDefault(sort = "name", size = 10) Pageable pageable){
      return testEntityRepository.findAll(pageable).getContent();
  }
```

**Comment by eiswind** [comment]
@bnasslahsen I do not see this in the docs. Maybe I'm blind :) I remember that this was mentioned there once. But in the text I can see now it says that it is supported out of the box.

Anyway it still does not work when I add the suggested dependency (which pulls in a lot of unwanted dependencies).

I updated the github example with your suggestions and the default value gets reported as 20. Did I still get anything wrong?

**Comment by bnasslahsen** [comment]
@eiswind, 

You are missing `@ParameterObject` ...
Just read the docs well and apply the guidelines:

1. You are using spring-data-rest types, so you have to add the right [dependency](https://springdoc.org/#spring-data-rest-support):  `springdoc-openapi-data-rest`
2. For [PageableDefault](https://springdoc.org/#is-pageabledefault-supported-to-enhance-the-openapi-3-docuementation), if you just read and apply what is writing in the docs: **use it in conjunction with @ParameterObject annotation. Also, the spring-boot spring.data.web. and spring.data.rest.default. properties are supported since v1.4.5**

Additionally, if you did some little efforts you would find a lot of samples in the project tests:
https://github.com/springdoc/springdoc-openapi/blob/0a8150b3158447e7ed88215e6bdeae2a7b75808b/springdoc-openapi-data-rest/src/test/java/test/org/springdoc/api/app13/HelloController.java#L35-L60

Not that next time, If you have a general usage question please ask on Stack Overflow. 
The springdoc-openapi team and the broader community monitor the `springdoc` tag.
We use GitHub issues to track bugs and enhancements.

Make sure you read the  [code of conduct.](https://github.com/springdoc/springdoc-openapi/blob/0a8150b3158447e7ed88215e6bdeae2a7b75808b/CODE_OF_CONDUCT.md) as we expect people to be at least respectful and constructive in their comments. 

The last thing, is if you believe there is a better url handling, feel free to propose a PR.

**Comment by eiswind** [comment]
@bnasslahsen I really did not intend to be offending. Please accept my excuses if you felt bad about my comments.
I really appreciate your help.

I must admit that I was overwhelmed by the many options and alternatives mentioned in the docs and I -really- thought it was a bug.

So once again sorry if you felt offended.

**Comment by bnasslahsen** [comment]
@eiswind, 

Note that after our last discussion, i came up with an optimization for the `PageableDefault` support, so it can be supported  (same as `Pageable`) out of the box, without the extra dependency `springdoc-openapi-data-rest`.

This enhancement will be available for the next release v1.6.4.
